uint16 state
uint16 pwm